src/mugs.js

```diff
@@ -649,6 +662,20 @@ define([
             return attrs.nodeset ? [attrs] : [];
         },
 
+        getSetValues: function (mug) {
+            var ret = [];
+
+            if (mug.p.setValue) {
+                ret = [{
+                    value: mug.p.setValue,
+                    event: mug.isInRepeat() ? 'jr-insert' : 'xforms-ready',
```

Comment:
Could this change the event if, for example, the form originally had `<setvalue event="jr-insert" .../>` for this mug but it was not in a repeat?